save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPluginConfig.kt

```diff
@@ -1,49 +1,64 @@
+@file:Suppress("HEADER_MISSING_IN_NON_SINGLE_CLASS_FILE")
+@file:UseSerializers(RegexSerializer::class)
+
 package org.cqfn.save.plugins.fix
 
 import org.cqfn.save.core.config.TestConfigSections
 import org.cqfn.save.core.plugin.PluginConfig
-
-import okio.Path
+import org.cqfn.save.core.utils.RegexSerializer
 
 import kotlinx.serialization.Serializable
+import kotlinx.serialization.UseSerializers
 
 /**
  * Some fields by default are null, instead of some natural value, because of the fact, that in stage of merging
  * of nested configs, we can't detect whether the value are passed by user, or taken from default.
  * The logic of the default value processing will be provided in stage of validation
  *
  * @property execCmd a command that will be executed to mutate test file contents
- * @property destinationFileSuffix [execCmd] should append this suffix to the file name after mutating it.
+ * @property resourceNameTestSuffix suffix name of the test file.
+ * @property resourceNameExpectedSuffix suffix name of the expected file.
  */
 @Serializable
 data class FixPluginConfig(
     val execCmd: String,
-    val destinationFileSuffix: String? = null,
+    val resourceNameTestSuffix: String? = null,
+    val resourceNameExpectedSuffix: String? = null,
 ) : PluginConfig {
     override val type = TestConfigSections.FIX
 
+    /**
+     *  @property resourceNamePattern regex for the name of the test files.
+     */
+    val resourceNamePattern: Regex = resourceNamePattern()
+
+    /**
+     *  @property resourceNameTest
+     */
+    val resourceNameTest: String = resourceNameTestSuffix ?: "Test"
+
+    /**
+     *  @property resourceNameExpected
+     */
+    val resourceNameExpected: String = resourceNameExpectedSuffix ?: "Expected"
+
     override fun mergeWith(otherConfig: PluginConfig): PluginConfig {
         val other = otherConfig as FixPluginConfig
         return FixPluginConfig(
             this.execCmd,
-            this.destinationFileSuffix ?: other.destinationFileSuffix
+            this.resourceNameTestSuffix ?: other.resourceNameTestSuffix,
+            this.resourceNameExpectedSuffix ?: other.resourceNameExpectedSuffix,
         )
     }
 
-    /**
-     * Constructs a name of destination file from original file name and [destinationFileSuffix]
-     *
-     * @param original an original file
-     * @return a name of destination file
-     */
-    fun destinationFileFor(original: Path): String {
-        requireNotNull(destinationFileSuffix) { "This method should never be called when destinationFileSuffix is null" }
-        return original.name.run {
-            substringBeforeLast(".") + destinationFileSuffix + "." + substringAfterLast(".")
+    private fun resourceNamePattern(): Regex {
+        val isNotNullResourceNameTestSuffix = resourceNameTestSuffix != null
```

Comment:
Can you use `resourceNameTest` here, instead of checking nullability once again?